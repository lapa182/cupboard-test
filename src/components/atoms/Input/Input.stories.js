import React from "react";

import { storiesOf } from "@storybook/react";
import { withKnobs, text } from "@storybook/addon-knobs";
import Input from ".";

const placeholderText = text("Alternative Text", "Filter value");

storiesOf("Input", module)
    .addDecorator(withKnobs)
    .add("default", () => <Input placeholder={placeholderText} />);
