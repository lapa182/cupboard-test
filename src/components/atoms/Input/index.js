import * as React from "react";
import PropTypes from "prop-types";

const Input = ({ placeholder, onKeyUp }) => (
  <div className="field">
    <div className="control">
      <input
        className="input"
        type="text"
        onKeyUp={onKeyUp}
        placeholder={placeholder}
        aria-label={placeholder}
      />
    </div>
  </div>
);

Input.propTypes = {
  placeholder: PropTypes.string,
  onKeyUp: PropTypes.func.isRequired
};

Input.defaultProps = {
  placeholder: "Filter value"
};

export default Input;
