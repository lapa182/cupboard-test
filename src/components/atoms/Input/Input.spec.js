import React from "react";
import ReactDOM from "react-dom";
import Input from "./index";

describe("Picture", () => {
  it("renders without crashing", () => {
    const picture = document.createElement("input");
    ReactDOM.render(<Input />, Input);
    ReactDOM.unmountComponentAtNode(picture);
  });
});
