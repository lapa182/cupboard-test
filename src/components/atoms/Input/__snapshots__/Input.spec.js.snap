// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders without crashing 1`] = `
<div
  className="field"
>
  <div
    className="control"
  >
    <input
      aria-label="Filter value"
      className="input"
      placeholder="Filter value"
      type="text"
    />
  </div>
</div>
`;
